scripts/bench_python_cpp_algos.py

```diff
@@ -61,199 +68,241 @@ def save_statistics_as_json(bench_stats, start_time, path):
+        3. If there are any steps that are required but unfulfilled add them
 
 
-def pagerank(graph: Graph, _input_args):
+    :name: The name of the routine
+    :graph: The graph to use
+    :input_args: The input arguments
+    :num_sources: The number of sources for BC
+    :source_node_file: The source file containing source nodes
+    :return: Benchmarking data dictionary
+    """
+    if name == "tc":
+        return tc(graph, "")
+
     property_name = "NewProp"
+    start_node = input_args["source_node"]
+    compare_node = input_args["source_node"]
+    edge_prop_name = input_args["edge_wt"]
 
     tolerance = 0.0001
     max_iteration = 1000
     alpha = 0.85
-    time_data = {}
-
-    with time_block(analytics.PagerankPlan.__name__, time_data):
-        pagerank_plan = analytics.PagerankPlan.pull_topological(tolerance, max_iteration, alpha)
-
-    with time_block(analytics.pagerank.__name__, time_data):
-        analytics.pagerank(graph, property_name, pagerank_plan)
-
-    with time_block(check_schema.__name__, time_data):
-        check_schema(graph, property_name)
-
-    analytics.pagerank_assert_valid(graph, property_name)
+    k = 10
 
-    stats = analytics.PagerankStatistics(graph, property_name)
-    print(f"STATS:\n{stats}")
-    with time_block(graph.remove_node_property.__name__, time_data):
-        graph.remove_node_property(property_name)
+    cc_bc_args = [graph, property_name]
+    k_args = [graph, k, property_name]
+    jaccard_args = [graph, compare_node, property_name]
+    bfs_args = [graph, start_node, property_name]
+    sssp_args = [graph, start_node, edge_prop_name, property_name]
+    louvain_args = [graph, edge_prop_name, property_name]
+    pagerank_args = [graph, property_name]
+    time_data = {}
+    func_arg_mapping = {
+        "cc": Routine(
+            RoutineFunc(
+                None,
+                analytics.connected_components,
+                analytics.connected_components_assert_valid,
+                analytics.ConnectedComponentsStatistics,
+            ),
+            RoutineArgs(None, cc_bc_args, cc_bc_args, cc_bc_args),
+        ),
+        "kcore": Routine(
+            RoutineFunc(None, analytics.k_core, analytics.k_core_assert_valid, analytics.KCoreStatistics),
+            RoutineArgs(None, k_args, k_args, k_args),
+        ),
+        "bfs": Routine(
+            RoutineFunc(
+                analytics.BfsPlan.asynchronous
+                if "road" in input_args["name"]
+                else analytics.BfsPlan.synchronous_direction_opt,
+                analytics.bfs,
+                analytics.bfs_assert_valid,
+                analytics.BfsStatistics,
+            ),
+            RoutineArgs([] if "road" in input_args["name"] else [15, 18], bfs_args, bfs_args, (graph, property_name)),
+        ),
+        "sssp": Routine(
+            RoutineFunc(
+                analytics.SsspPlan.delta_step
+                if "kron" in input_args["name"] or "urand" in input_args["name"]
```

Comment:
minor: should this `if` be on the previous line as it reads nicer

Reply:
Unfortunately this gets overridden by format fix.